superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/dndControls.tsx

```diff
@@ -178,14 +193,30 @@ export const dnd_granularity_sqla: typeof dndGroupByControl = {
       : 'Drop temporal column here',
   ),
   mapStateToProps: ({ datasource }) => {
-    const temporalColumns = datasource?.columns.filter(c => c.is_dttm) ?? [];
+    if (datasource?.columns[0]?.hasOwnProperty('column_name')) {
+      const temporalColumns =
+        (datasource as Dataset)?.columns.filter(c => c.is_dttm) ?? [];
+      const options = Object.fromEntries(
+        temporalColumns.map(option => [option.column_name, option]),
+      );
+      return {
+        options,
+        default:
+          (datasource as Dataset)?.main_dttm_col ||
+          temporalColumns[0]?.column_name ||
+          null,
+        isTemporal: true,
+      };
+    }
+
+    const temporalColumns =
+      (datasource as QueryResponse)?.columns.filter(c => c.is_dttm) ?? [];
```

Comment:
This is the one where let's show all the columns but put the is_dttm ones at the top, since users can't turn on/ off the is_dttm property.

Reply:
Ohhh yes that's right. I added a sort for this in [`this commit`](https://github.com/apache/superset/pull/19855/commits/ac671a4e7f65ce966a1d2c8d9f637d5bfb8cf64a).